examples/example/src/main/java/cz/smarteon/loxone/example/LoxoneJavaExample.java

```diff
@@ -1,56 +1,98 @@
-        loxoneConfig.getControls().entrySet().stream().findFirst().ifPresent(e -> {
+        try {
+            // Start the service.
+            loxone.start();
 
-            final String controlId = e.getKey().toString();
-            final CountDownLatch latch = new CountDownLatch(1);
+            // Wait for command and any event.
+            commandLatch.await();
+            eventLatch.await();
 
-            // set listener waiting for our command
-            loxoneWebSocket.registerListener((command, value) -> {
-                System.out.println("Got answer on command=" + command + " value=" + value);
-                if (command.contains(controlId)) {
-                    latch.countDown();
-                }
-                return CommandListener.State.CONSUMED;
-            });
+        } catch (InterruptedException | LoxoneException e) {
+            e.printStackTrace();
+        }
 
-            // send the command
-            loxoneWebSocket.sendCommand(Protocol.jsonControlAll(controlId));
+        // Release all the resources correctly.
+        loxone.stop();
+    }
 
-            try {
-                latch.await();
-            } catch (InterruptedException e1) {
-                e1.printStackTrace(); //ugly :(
+    private static void setEventListener(Loxone loxone, CountDownLatch latch) {
+        loxone.setEventsEnabled(true);
+        loxone.webSocket().registerListener(new LoxoneEventListener() {
+            @Override
+            public void onEvent(@NotNull final ValueEvent event) {
+                latch.countDown();
+                System.out.println("Received event=" + event);
+            }
+
+            @Override
+            public void onEvent(@NotNull final TextEvent event) {
+                latch.countDown();
+                System.out.println("Received event=" + event);
             }
         });
+    }
 
-        loxoneWebSocket.close();  // websocket lib creates daemon threads so we need to close explicitly
+    private static void setCommandResponseListener(Loxone loxone, CountDownLatch latch) {
+        loxone.webSocket().registerListener(new CommandResponseListener<LoxoneMessage>() {
+            @Override
+            @NotNull
+            public State onCommand(@NotNull final Command<? extends LoxoneMessage> command, @NotNull final LoxoneMessage message) {
+                System.out.println("Got answer on command=" + command + " message=" + message);
+                latch.countDown();
+                return State.CONSUMED;
+            }
+
+            @Override
+            public boolean accepts(@NotNull final Class clazz) {
+                return LoxoneMessage.class.equals(clazz);
+            }
+        });
+    }
+
+    private static void setLoxoneAppListener(Loxone loxone) {
+        loxone.registerLoxoneAppListener(loxoneApp -> {
+            // Obtain any Control and send the command.
+            loxoneApp.getControls(SwitchControl.class).stream().findFirst().ifPresent(
```

Comment:
In case there is no switch all the example does nothing and waits forever (I guess), could you at least printout what happened?